app/src/test/java/ch/hikemate/app/authentication/AuthViewModelTest.kt

```diff
@@ -0,0 +1,144 @@
+
+  @Mock private lateinit var mockFirebaseUser: FirebaseUser
+
+  private lateinit var staticFirebaseApp: MockedStatic<FirebaseApp>
+
+  private lateinit var staticFirebaseAuth: MockedStatic<FirebaseAuth>
+
+  // The ViewModel under test
+  private lateinit var viewModel: AuthViewModel
+
+  @Before
+  fun setup() {
+    mockRepository = mock()
+    mockContext = mock()
+    mockFirebaseAuth = mock()
+    mockFirebaseUser = mock()
+
+    staticFirebaseApp = mockStatic(FirebaseApp::class.java)
+    staticFirebaseAuth = mockStatic(FirebaseAuth::class.java)
+
+    // Define the behavior of FirebaseApp and FirebaseAuth when called.
+    // This is necessary because the AuthViewModel calls and uses on FirebaseAuth.
+    `when`(FirebaseApp.initializeApp(any<Context>())).thenReturn(mock(FirebaseApp::class.java))
+    `when`(FirebaseAuth.getInstance()).thenReturn(mockFirebaseAuth)
+  }
+
+  // Helper functions to setup the ViewModel with a signed-in or signed-out user
+  private fun setupSignedInUser() {
+    `when`(mockFirebaseAuth.currentUser).thenReturn(mockFirebaseUser)
+    viewModel = AuthViewModel(mockRepository)
+  }
+
+  // Helper functions to setup the ViewModel with a signed-in or signed-out user
+  private fun setupSignedOutUser() {
+    `when`(mockFirebaseAuth.currentUser).thenReturn(null)
+    viewModel = AuthViewModel(mockRepository)
+  }
+
+  @After
+  fun tearDown() {
+    staticFirebaseApp.close()
+    staticFirebaseAuth.close()
+  }
+
+  @Test
+  fun signInWithGoogle_calls_repository_and_updates_currentUser_on_success() = runTest {
+    setupSignedOutUser()
+
+    // Simulate a successful Google sign-in by invoking the onSuccess callback
+    doAnswer { arguments ->
+          val onSuccess = arguments.getArgument<(FirebaseUser?) -> Unit>(0)
+          onSuccess(mockFirebaseUser) // Call the success callback with a mock user
+          null
+        }
+        .`when`(mockRepository)
+        .signInWithGoogle(any(), any(), any(), any(), any())
+
+    // Verify that currentUser is initially null
+    assertNull(viewModel.currentUser.first())
+
+    viewModel.signInWithGoogle(this, mockContext)
+
+    val currentUser = viewModel.currentUser.first() // Get the first (current) value of the flow
+    assertEquals(mockFirebaseUser, currentUser)
+  }
+
+  @Test
+  fun signInWithGoogle_calls_repository_and_does_not_update_currentUser_on_error() = runTest {
+    setupSignedOutUser()
+
+    // Simulate an unsuccessful Google sign-in by invoking the onError callback
+    doAnswer { invocation ->
+          val onErrorAction = invocation.getArgument<(Exception) -> Unit>(1)
+          onErrorAction(Exception("Error"))
+          null
+        }
+        .`when`(mockRepository)
+        .signInWithGoogle(any(), any(), any(), any(), any())
+
+    // Verify that currentUser is initially null
+    assertNull(viewModel.currentUser.first())
+
+    viewModel.signInWithGoogle(this, mockContext)
+
```

Comment:
I think you forgot to check if the function calls the repository here. You did it in the sign out test.
I suppose you should add something like `verify(mockRepository).signInWithGoogle(any(),any(),any(),any(),any())` or something like that.